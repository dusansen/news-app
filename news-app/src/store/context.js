import React, { createContext, useReducer, useContext } from 'react';
import { reducer } from './reducer';
import { COUNTRIES } from '../utils/constants';

const initialState = {
  country: COUNTRIES[0]
};

const context = createContext(initialState);
const { Provider } = context;

const StateProvider = ({ children }) => {
  const [state, dispatch] = useReducer(reducer, initialState);

  return <Provider value={{ state, dispatch }}>{children}</Provider>
};

const useAppContext = () => useContext(context);

export { context, StateProvider, useAppContext };