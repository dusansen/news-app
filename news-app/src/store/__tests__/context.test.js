import { context } from '../context';

describe('Context', () => {
  it('should match context initial state snapshot', () => {
    expect(context).toMatchSnapshot();
  });
});