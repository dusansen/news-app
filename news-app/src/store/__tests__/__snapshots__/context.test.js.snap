// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Context should match context initial state snapshot 1`] = `
Object {
  "$$typeof": Symbol(react.context),
  "Consumer": Object {
    "$$typeof": Symbol(react.context),
    "_calculateChangedBits": null,
    "_context": [Circular],
  },
  "Provider": Object {
    "$$typeof": Symbol(react.provider),
    "_context": [Circular],
  },
  "_calculateChangedBits": null,
  "_currentRenderer": null,
  "_currentRenderer2": null,
  "_currentValue": Object {
    "articles": Object {},
    "country": Object {
      "label": "Great Britain",
      "value": "GB",
    },
    "currentArticle": null,
    "errorMessage": "",
  },
  "_currentValue2": Object {
    "articles": Object {},
    "country": Object {
      "label": "Great Britain",
      "value": "GB",
    },
    "currentArticle": null,
    "errorMessage": "",
  },
  "_threadCount": 0,
}
`;
